### DISCUSSION TITLE
Cannot read multidimensional postgresql array: Expected to read a Postgres list value of length 4, but only have size 2

### DISCUSSION THREAD

**Discussion by kamilcglr** [duckdb/duckdb#7066] [Q&A] [answered]
Hello,

I can't read a postgresql table where I have a two dimensional array column.
Here is a simple example to reproduce the problem.

_In postgres database_
```sql
CREATE TABLE test
(
    ids INTEGER[][]
);

INSERT INTO test (ids)
VALUES (ARRAY [1,2,3]);
```

_In duckdb_
```sql
INSTALL
postgres_scanner;
    LOAD
postgres_scanner;
CALL postgres_attach(***);

-- Then
SELECT ids::INT[] FROM test; -- This works
```

But if I insert a two-dimensional array in postgres
```sql
INSERT INTO test (ids)
VALUES (ARRAY [ARRAY [1,2], ARRAY [3, 4]]);
```

I have the following error in duckdb
```sql
SELECT ids::INT[] FROM test;
Error: Invalid Error: Invalid Input Error: Expected to read a Postgres list value of length 4, but only have size 2
```

I tried to cast it to varchar, but same error
```sql
SELECT CAST (ids AS VARCHAR) FROM test;
Error: Invalid Error: Invalid Input Error: Expected to read a Postgres list value of length 4, but only have size 2

-- same for 
SELECT ids::int[][] FROM test;
Error: Invalid Error: Invalid Input Error: Expected to read a Postgres list value of length 4, but only have size 2
```

I found this [issue](https://github.com/duckdb/duckdb/issues/2698#issuecomment-1486574567) that explains that duckdb is able to convert a string into an array and indeed it works
```sql
SELECT '[[1,2],[3,4]]'::INT[][] AS l;
```

I have the impression that this is a problem with postgresql.

Thank you in advance for your help

**Comment by Mytherin** [comment, marked as answer]
I believe the Postgres scanner currently doesn't support multi-dimensional arrays, have a look at the [Postgres scanner repo](https://github.com/duckdblabs/postgres_scanner/issues)

**Reply by kamilcglr** [reply]
Thank you, is there an alternative?

**Reply by kamilcglr** [reply]
We are going to use `jsonb` type
